x-pack/plugins/observability_solution/entity_manager/server/lib/entities/built_in/services.ts

```diff
@@ -25,42 +34,82 @@ export const builtInServicesEntityDefinition: EntityDefinition = entityDefinitio
   displayNameTemplate: '{{service.name}}{{#service.environment}}:{{.}}{{/service.environment}}',
   metadata: [
     'data_stream.type',
-    'service.instance.id',
     'service.namespace',
     'service.version',
     'service.runtime.name',
     'service.runtime.version',
     'service.node.name',
     'service.language.name',
-    'agent.name',
```

Comment:
is agent name considered to be high cardinality?

Reply:
FYI, we need the `agent.name` field in the inventory to display the agent icon.

Reply:
Reading [this data](https://github.com/elastic/elastic-entity-model/issues/30#issuecomment-2189270323) from SRE overview cluster the cardinality of agent.name is the same as host.name or instance.id. I could see this happening for agents living in the same container as the monitored service, restarting the whole container would spawn new agents as well. If this is mandatory information we should probably take the risk to include it; cc @tommyers-elastic

Reply:
@kpatticha where is the agent icon displayed in inventory ? I can't find it on edge cluster

Reply:
is the little icon next to the service name.

<img width="289" alt="image" src="https://github.com/elastic/kibana/assets/3369346/1f10238e-1f7a-4388-ba91-6383eb6037e4">
 
  we also display the icon when you click the service name and open the service details page.

Reply:
@kpatticha which fields do we use to display docker or kubernetes icon in service overview page?

Reply:
Thanks, so the `agent.name` carries the language information for apm. Could we rely on `service.language.name` instead ? Values would be different so inventory logic have to be slightly updated but we won't have risks of high cardinality for this field

Reply:
For  kubernetes icon we check first for the field `kubernetes.pod.name` 
For the container just `container.id`

if the document has both, we display the kubernetes icon

Reply:
are we making changes to the existing code on the service overview page as part of this work ?

Reply:
No we don't change this specific functionality now and the logs-only service overview page doesn't include the icons but in the future it would be useful to fetch this metadata from the entities.